lib/std/start.zig

```diff
@@ -147,6 +147,17 @@ fn _start() callconv(.Naked) noreturn {
                 : "r0"
             );
         },
+        .sparcv9 => {
+            // On sparc64, the stack pointer register points to a place
+            // 2047 bytes below the actual stack. Also, argc and friends are
+            // placed starting at [stack-start + 128], so we need to account for that too.
+            // Ref: System V Application Binary Interface: SPARC Version 9 Processor Supplement
+            // Version 1.35, figure 3-16.
+            // TODO: find a better way to do this.
```

Comment:
Define "better". 
The formula is `o6` plus the stack bias (0x7ff) and the reserved area (16 64bit register = 0x80), you can either do the calculation in the asm block or return the sp and perform the addition in using Zig code.

You should also add a `mov %%g0, %%fp` (tip: you can use `%sp`/`%fp` instead of `%o7`/`%i7` to terminate the frame pointer chain.

Reply:
I... dunno. Having a variable named `starting_stack_ptr` points to 128 bytes past the stack top (or bottom?) just feels somewhat hacky to me. Then again, it seems to be only used to extract argc/argv...

Reply:
Oh I see, the name is misleading. Any idea for a better name? I'd suggest something like `argc_argv_ptr`.

Reply:
Mhm, changed it to better reflect the intent. `argc_argv_ptr` is fine since it's the pointer's only purpose, I think.